ui/src/components/layouts/sider/LayoutSider.tsx

```diff
@@ -3,31 +3,71 @@
  * SPDX-FileCopyrightText: Huawei Inc.
  */
 
-import { Image, Menu } from 'antd';
-import Sider from 'antd/es/layout/Sider';
-import { useState } from 'react';
-import { Link } from 'react-router-dom';
-import { homePageRoute, usernameKey } from '../../utils/constants';
+import { Image, Menu, Layout } from 'antd';
+import {useEffect, useState} from 'react';
+import {Link, useNavigate} from 'react-router-dom';
+import {homePageRoute, usernameKey} from '../../utils/constants';
 import registerPanelMenu from '../../content/register/registerPanelMenu';
-import { MenuItemType } from 'antd/lib/menu/hooks/useItems';
+import {ItemType} from "antd/es/menu/hooks/useItems";
+import {serviceVendorApi} from "../../../xpanse-api/xpanseRestApiClient";
+import {AppstoreAddOutlined, BarsOutlined} from "@ant-design/icons";
+
+const mapper: any = {
+  ai: 'AI',
+  compute: 'Compute',
+  media_service: 'MediaService',
+  middleware: 'MiddleWare',
+  database: 'Database',
+  container: 'Container',
+  storage: 'Storage',
+  network: 'Network',
+  security: 'Security',
+  others: 'Others',
+};
 
 function LayoutSider(): JSX.Element {
     const [collapsed, setCollapsed] = useState(false);
-    const menuItems: MenuItemType[] = [];
+    const [items, setItems] = useState<ItemType[]>([]);
+    const navigate = useNavigate();
+    const onSelected = function (cfg: any) {
+        navigate(cfg.key);
+    };
+
+    const onClicked = function (cfg: any) {
+        navigate(cfg.key);
+    };
 
-    if (localStorage.getItem(usernameKey) === 'csp') {
-        menuItems.push(registerPanelMenu());
-    }
+    useEffect(() => {
+        serviceVendorApi.listCategories().then((rsp) => {
+            const data = rsp;
+            const items = data.map((subMenu: string) => {
+                return {
+                    key: '/catalog#' + subMenu,
+                    label: mapper[subMenu] || subMenu,
+                };
+            });
+          if (localStorage.getItem(usernameKey) === 'csp') {
+            setItems([
+                {
```

Comment:
We can move this part to the Catalog component like the `registerPanelMenu` do.

Reply:
fixed.